nntrainer/layers/layer_node.h

```diff
@@ -337,7 +364,8 @@ class LayerNode : public ml::train::Layer, public GraphNode {
    * These properties are set for the layer by the user but are intercepted
    * and used in the node which forms the basic element of the graph.
    */
-  std::tuple<props::Name> props; /**< properties for the layer node */
+  std::unique_ptr<std::tuple<props::Name>>
```

Comment:
does this need unique_ptr?

Reply:
This is to pimple the props, unique_ptr allows incomplete type. Props are quite heavy to compile but overhead from indirection is negligible.

Reply:
sounds good.

Reply:
is it ok when we need to copy the properties?

Reply:
Yes, copy is implemented for properties.